src/test/scala/de/frosner/ddq/CheckTest.scala

```diff
@@ -120,172 +146,280 @@ class CheckTest extends FlatSpec with Matchers with BeforeAndAfterEach with Befo
+    val constraint = check.constraints.head
+    val result = ConstraintFailure("Column column contains 2 rows that are not in Set(a, b, d)")
+    check.run().constraintResults shouldBe Map(constraint -> result)
   }
 
   "A check if a column satisfies the given regex" should "succeed if all values satisfy the regex" in {
-    Check(makeNullableStringDf(List("Hello A", "Hello B", "Hello C"))).isMatchingRegex("column", "^Hello").run shouldBe true
+    val check = Check(makeNullableStringDf(List("Hello A", "Hello B", "Hello C"))).isMatchingRegex("column", "^Hello")
+    val constraint = check.constraints.head
+    val result = ConstraintSuccess("Column column matches ^Hello")
+    check.run().constraintResults shouldBe Map(constraint -> result)
   }
 
   it should "succeed if all values satisfy the regex or are null" in {
-    Check(makeNullableStringDf(List("Hello A", "Hello B", null))).isMatchingRegex("column", "^Hello").run shouldBe true
+    val check = Check(makeNullableStringDf(List("Hello A", "Hello B", null))).isMatchingRegex("column", "^Hello")
+    val constraint = check.constraints.head
+    val result = ConstraintSuccess("Column column matches ^Hello")
+    check.run().constraintResults shouldBe Map(constraint -> result)
   }
 
   it should "fail if there are values not satisfying the regex" in {
-    Check(makeNullableStringDf(List("Hello A", "Hello B", "Hello C"))).isMatchingRegex("column", "^Hello A$").run shouldBe false
-  }
-
-  "Multiple checks" should "fail if one check is failing" in {
-    Check(makeIntegerDf(List(1,2,3))).hasNumRowsEqualTo(3).hasNumRowsEqualTo(2).run shouldBe false
-  }
-
-  it should "succeed if all checks are succeeding" in {
-    Check(makeIntegerDf(List(1,2,3))).hasNumRowsEqualTo(3).hasUniqueKey("column").satisfies("column > 0").run shouldBe true
+    val check = Check(makeNullableStringDf(List("Hello A", "Hello B", "Hello C"))).isMatchingRegex("column", "^Hello A$")
+    val constraint = check.constraints.head
+    val result = ConstraintFailure("Column column contains 2 rows that do not match ^Hello A$")
+    check.run().constraintResults shouldBe Map(constraint -> result)
+  }
+
+  "Multiple checks" should "produce a constraintResults map with all constraints and corresponding results" in {
+    val check = Check(makeIntegerDf(List(1,2,3))).hasNumRowsEqualTo(3).hasNumRowsEqualTo(2).satisfies("column > 0")
+    val constraint1 = check.constraints(0)
+    val constraint2 = check.constraints(1)
+    val constraint3 = check.constraints(2)
+
+    check.run().constraintResults shouldBe Map(
+      constraint1 -> ConstraintSuccess("The number of rows is equal to 3"),
+      constraint2 -> ConstraintFailure("The actual number of rows 3 is not equal to the expected 2"),
+      constraint3 -> ConstraintSuccess("Constraint column > 0 is satisfied")
+    )
   }
 
   "It" should "be possible to specify a display name for a data frame" in {
```

Comment:
@Gerrrr Actually, this test can now be improved. It does no actually test the table name output so I will either add this to the `RunnerTest` or create a new integration test that checks if the table name is put as a header in the `CheckResult`